Community-Supported/git-to-hyper/advanced-git-to-hyper.py

```diff
@@ -0,0 +1,271 @@
+"""
+This file serves as an example on how to insert data into a hyper database while using multiple
+processes to extract data at the same time.
+
+As the Hyper API does not allow multiple connections to the same database it is not possible to
+insert the data directly from all processes. To overcome this limitation the following example uses
+multiple "extraction processes" mining the data and a single "Injection Process" which inserts it
+into the Hyper database. The data is synced between those processes using one Manager.Queue object
+from the multiprocessing library per SQL Table.
+
+This way only a single connection to the Hyper database is required while all available resources
+can be used to extract the data.
+"""
+
+import click
+import queue
+import os
+import tempfile
+import hashlib
+import subprocess
+import time
+
+import tableauhyperapi as hapi
+
+from distutils.dir_util import copy_tree
+from pathlib import Path
+from git import Repo
+from git.objects import Commit
+from multiprocessing import cpu_count, Manager, Process
+from typing import List, Dict
+from enum import Enum
+
+__database_file = Path("git.hyper")
+
+
+class Tables(Enum):
+    COMMITS = 0,
+    CHANGED_FILES = 1,
+    FILE_COMMIT_MAPPING = 2,
+    BLAME = 3
+
+
+def set_up_database(con: hapi.Connection) -> Dict[str, hapi.TableDefinition]:
+    tables = dict()
+    tables[Tables.COMMITS] = create_table(con, 'commits', [
```

Comment:
please order the functions in this file in a logical order. Very surprising that `create_table` is only defined much later